### DISCUSSION TITLE
How to filter channels by custom fields?

### DISCUSSION THREAD

**Discussion by Brazol** [GetStream/stream-chat-flutter#797] [Q&A] [answered]
Hi, in docs about quering channels (https://getstream.io/chat/docs/react/querying_channels/?language=dart) its stated to `Use custom fields on channels to make filters more simple.`. How to accomplish that? I tried something like that but it doesn't work:

` const Filter.raw(value: {
            'extraData': {
              'customFields': {'someField': 'someValue'}
            }
          })`

**Comment by imtoori** [comment, marked as answer]
you shouldn't use `extraData`

for example
```dart
const Filter.raw(value: {'myfield': true})
```

**Reply by Brazol** [reply]
what if I have map inside extraData? Like we have field 'customFields' that is a map. Can I filter based on field inside 'customFields'?

something like this causes 500 response
`const Filter.raw(value:{ 'customFields': {'someField': 'someValue'} } )`

**Reply by imtoori** [reply]
yeah I don't think you can do that
